includes/class-webmention-receiver.php

```diff
@@ -264,47 +311,42 @@ public static function post( $request ) {
 		}
 		// In the event of async processing this needs to be stored here as it might not be available
 		// later.
-		$comment_meta                          = array();
-		$comment_author_ip                     = preg_replace( '/[^0-9a-fA-F:., ]/', '', $_SERVER['REMOTE_ADDR'] );
-		$comment_agent                         = isset( $_SERVER['HTTP_USER_AGENT'] ) ? $_SERVER['HTTP_USER_AGENT'] : '';
-		$comment_date                          = current_time( 'mysql' );
-		$comment_date_gmt                      = current_time( 'mysql', 1 );
-		$comment_meta['webmention_created_at'] = $comment_date_gmt;
-
-		if ( isset( $params['vouch'] ) ) {
-			// If there is a vouch pass it along
-			$vouch = urldecode( $params['vouch'] );
-			// Safely store a version of the data
-			$comment_meta['webmention_vouch_url'] = esc_url_raw( $vouch );
-		}
+		$comment_author_ip = preg_replace( '/[^0-9a-fA-F:., ]/', '', $_SERVER['REMOTE_ADDR'] );
+		$comment_agent     = isset( $_SERVER['HTTP_USER_AGENT'] ) ? $_SERVER['HTTP_USER_AGENT'] : '';
+		$comment_date      = current_time( 'mysql' );
+		$comment_date_gmt  = current_time( 'mysql', 1 );
+		$created           = current_time( DATE_W3C, 1 );
+		$protocol          = 'webmention';
 
 		// change this if your theme can't handle the Webmentions comment type
 		$comment_type = WEBMENTION_COMMENT_TYPE;
 
 		// change this if you want to auto approve your Webmentions
 		$comment_approved = WEBMENTION_COMMENT_APPROVE;
 
-		$commentdata = compact( 'comment_type', 'comment_approved', 'comment_agent', 'comment_date', 'comment_date_gmt', 'comment_meta', 'source', 'target', 'vouch' );
+		$commentdata = compact( 'protocol', 'comment_type', 'comment_approved', 'comment_agent', 'comment_date', 'comment_date_gmt', 'created', 'source', 'target', 'vouch' );
```

Comment:
do we still need this? I thought this was only to prefill the array, but with https://github.com/pfefferle/wordpress-webmention/pull/224/files#diff-4a405db849c60f60b9e32bde3bfe0096R232 shouldn't it be obsolete?

Reply:
Actually no. This still prefills the array. The store stores extra parameters in meta...the ones not stored in the comment table. Previously, our extra parameters were stored at the beginning rather than the end.

Reply:
Any other questions?

Reply:
But wasn't your initial problem, that `compact` does not work in newer PHP versions?

Reply:
There were two. The other was allowing more flexibility in the storage of parameters.

Reply:
The compact pre-fills the array with properties. The storing of them in meta, the ones that aren't part of the comment table, is now done in the new function.